cunumeric/module.py

```diff
@@ -7330,6 +7330,68 @@ def mean(
     )
 
 
+# weighted average
+
+
+@add_boilerplate("a", "weights")
+def average(
+    a: ndarray,
+    axis: Optional[Union[int, tuple[int, ...]]] = None,
+    weights: Union[ndarray, None] = None,
+    returned: bool = False,
+    *,
+    keepdims: bool = False,
+) -> Union[
+    Union[int, ndarray], Tuple[Union[int, ndarray], Union[int, ndarray]]
+]:
+    clean_axis: Optional[tuple[int, ...]] = None
+    if axis is not None:
+        clean_axis = normalize_axis_tuple(axis, a.ndim, argname="axis")
+
+    scl: Union[float, int, ndarray] = 1
+    if weights is None:
+        scl = float(
+            a.size
+            if clean_axis is None
+            else math.prod([a.shape[i] for i in clean_axis])
+        )
```

Comment:
Following the documentation of the function, let's only explicitly cast to `float64` if the array has integral type. Otherwise you may end up needlessly widening the result, if say the array was of type `np.float16`.

```suggestion
        scl = (
            a.size
            if clean_axis is None
            else math.prod([a.shape[i] for i in clean_axis])
        )
        if a.dtype.kind == "i":
            scl = np.float64(scl)
```